### DISCUSSION TITLE
TLS certs and cEOS

### DISCUSSION THREAD

**Discussion by networkop** [srl-labs/containerlab#857] [Q&A]
hey 👋 , I need a couple of self-signed certs for EOS's [restconf interface](https://aristanetworks.github.io/openmgmt/configuration/restconf/#certificate).
Is there an example of how to orchestrate this with containerlab?

**Comment by hellt** [comment]
Hi, mate
Today we generate a CA per lab deployment and keys for SR Linux as the latter needs TLS for gnmi to work.
If you deploy a lab you will see it under `clab-<labname>/ca` dir.

We can create client certs for ceos as well, it was just not needed by default as ceos supports insecure gnmi mode.

If you are okay with a semi-manual approach, then you can leverage https://containerlab.dev/cmd/tools/cert/sign/ command to create client cert/key for arista?

**Reply by hellt** [reply]
if you want to call this command after the lab is deployed, you can use `clab exec` for that or leverage `exec` node property - https://containerlab.dev/manual/nodes/#exec and make `Cli -p 15 "tls blah"` to make ceos generate self-signed cert on boot


I also think that maybe it makes sense to add this cert generation part of the ceos.Postdeploy stage, and, say, create `clab` key and cert for all ceos nodes.
If one happens to need a tls cert, it will be available by default

**Comment by networkop** [comment]
Yep, I noticed that. Is there anything special in `ceos` node kind to make it mount those files in the right directory?
I believe it doesn't use `/mnt/flash` but uses `/persist/secure` instead.

**Reply by hellt** [reply]
to make those files available (without asking user to configure `binds` in clab file) at certain paths one would need to add the right bind mounts here https://github.com/srl-labs/containerlab/blob/f918bdc11bfb281b056bde24c5fbea02efc24ad5/nodes/ceos/ceos.go#L70-L94

**Reply by networkop** [reply]
ACK.
